pkg/utils/helpers.go

```diff
@@ -350,14 +350,15 @@ func IsOwnedByCMA(addon *addonapiv1alpha1.ManagedClusterAddOn) bool {
 	return false
 }
 
-// GetSpecHash returns the sha256 hash of the spec field of the given object
+// GetSpecHash returns the sha256 hash of the spec field or data field of the given object
 func GetSpecHash(obj *unstructured.Unstructured) (string, error) {
 	if obj == nil {
 		return "", fmt.Errorf("object is nil")
 	}
 	spec, ok := obj.Object["spec"]
 	if !ok {
-		return "", fmt.Errorf("object has no spec field")
+		// handle cases like secret and configmaps
+		spec = obj.Object["data"]
```

Comment:
what if an API that does not have data field and spec field?

Reply:
var spec will be nil and return a hash like [case](https://github.com/open-cluster-management-io/addon-framework/pull/243/files#diff-e5c8046fa2e55ccd0b73a7f6cd75cfd2798f8e7bec08210444e01d6ffac0c0f0R91) .For example, if a configmap removed data field, hash will update and trigger an upgrade.

Reply:
how about we remove status and metadata part and hash?

Reply:
Yes, sounds better. Have modified the code.